adam_core/coordinates/tests/test_transforms_spice.py

```diff
@@ -0,0 +1,242 @@
+
+    diff = cartesian_elements_actual - cartesian_elements_spice
+
+    # Calculate offset in position in mm
+    r_diff = np.linalg.norm(diff[:, :3], axis=1) * u.au.to(u.mm)
+    # Calculate offset in velocity in nm/s
+    v_diff = np.linalg.norm(diff[:, 3:], axis=1) * (u.au / u.d).to(u.nm / u.s)
+
+    # Assert positions are to within 10 mm
+    np.testing.assert_array_less(r_diff, 10)
+    # Assert velocities are to within 10 nm/s
+    np.testing.assert_array_less(v_diff, 10)
+
+
+def test_keplerian_to_cartesian_hyperbolic_against_spice(orbital_elements):
+    # Test keplerian_to_cartesian (vmapped) against the cartesian elements
+    # calculated by SPICE for a series of sample orbital elements.
+
+    # Limit to hyperbolic orbits
+    orbital_elements = orbital_elements[orbital_elements["e"] > 1]
+
+    keplerian_elements = orbital_elements[["a", "e", "incl", "Omega", "w", "M"]].values
+    t0 = orbital_elements["mjd_tdb"].values
+
+    cartesian_elements_actual = keplerian_to_cartesian(keplerian_elements)
+
+    # Keplerian elements for SPICE are expected to be defined with periapse distance
+    # not the semi-major axis
+    t0_jd = t0 + 2400000.5
+    keplerian_elements_spice = np.empty((len(keplerian_elements), 8))
+    keplerian_elements_spice[:, 0] = orbital_elements["q"].values
+    keplerian_elements_spice[:, 1] = orbital_elements["e"].values
+    keplerian_elements_spice[:, 2] = np.radians(orbital_elements["incl"].values)
+    keplerian_elements_spice[:, 3] = np.radians(orbital_elements["Omega"].values)
+    keplerian_elements_spice[:, 4] = np.radians(orbital_elements["w"].values)
+    keplerian_elements_spice[:, 5] = np.radians(orbital_elements["M"].values)
+    keplerian_elements_spice[:, 6] = t0_jd
+    keplerian_elements_spice[:, 7] = MU
+
+    cartesian_elements_spice = np.empty((len(keplerian_elements), 6))
+    for i in range(len(keplerian_elements)):
+        cartesian_elements_spice[i] = sp.conics(keplerian_elements_spice[i], t0_jd[i])
+
+    diff = cartesian_elements_actual - cartesian_elements_spice
+
+    # Calculate offset in position in mm
+    r_diff = np.linalg.norm(diff[:, :3], axis=1) * u.au.to(u.mm)
+    # Calculate offset in velocity in nm/s
+    v_diff = np.linalg.norm(diff[:, 3:], axis=1) * (u.au / u.d).to(u.nm / u.s)
+
+    # Assert positions are to within 10 mm
+    np.testing.assert_array_less(r_diff, 10)
+    # Assert velocities are to within 10 nm/s
+    np.testing.assert_array_less(v_diff, 10)
+
+
+def test_cartesian_to_keplerian_elliptical_against_spice(orbital_elements):
+    # Test cartesian_to_keplerian (vmapped) against the keplerian elements
+    # calculated by SPICE for a series of sample orbital elements.
+
+    # Limit to elliptical orbits
+    orbital_elements = orbital_elements[orbital_elements["e"] < 1]
+
+    t0 = orbital_elements["mjd_tdb"].values
+    cartesian_elements = orbital_elements[["x", "y", "z", "vx", "vy", "vz"]].values
+    t0_jd = t0 + 2400000.5
```

Comment:
Not sure if it accounts for any of our disagreements, but floating point math here will stop working at a precision of about `0.0000000001`.